arrows/ffmpeg/ffmpeg_video_input.cxx

```diff
@@ -49,65 +49,43 @@ class ffmpeg_video_input::priv
 {
 public:
   /// Constructor
-  priv() :
-    f_format_context(avformat_alloc_context()),
-    f_video_index(-1),
-    f_video_encoding(nullptr),
-    f_video_stream(nullptr),
-    f_frame(nullptr),
-    f_filtered_frame(nullptr),
-    f_packet(nullptr),
-    f_software_context(nullptr),
-    f_filter_graph(nullptr),
-    f_filter_sink_context(nullptr),
-    f_filter_src_context(nullptr),
-    f_start_time(-1),
-    f_backstep_size(-1),
-    f_frame_number_offset(0),
-    video_path(""),
-    filter_desc("yadif=deint=1"),
-    frame_advanced(false),
-    end_of_video(true),
-    number_of_frames(0),
-    collected_all_metadata(false),
-    estimated_num_frames(false),
-    sync_metadata(true)
+  priv()
   { }
 
   // f_* variables are FFmpeg specific
 
-  AVFormatContext* f_format_context;
-  int f_video_index;
-  AVCodecContext* f_video_encoding;
-  AVStream* f_video_stream;
-  AVFrame* f_frame;
-  AVFrame* f_filtered_frame;
-  AVPacket* f_packet;
-  SwsContext* f_software_context;
-  AVFilterGraph* f_filter_graph;
-  AVFilterContext *f_filter_sink_context;
-  AVFilterContext *f_filter_src_context;
+  AVFormatContext* f_format_context = avformat_alloc_context();
```

Comment:
So, uh... not new to this branch, but this looks dodgy. This is freed by `priv::close`, and there are several paths that might call `priv::close` other than `~ffmpeg_video_input`... `ffmpeg_video_input::open` in particular.

...although on *deeper* investigation, it seems that a) `avformat_open_input` will create the context anyway, and so the usefulness of creating it earlier is questionable, and b) the call to `avformat_free_context` in `priv::close` will always be a no-op because the context is freed and set to null by the immediately preceding call to `avformat_close_input`. At the very least, I think the context needs to be explicitly nulled or recreated after calling `avformat_free_context`. (My vote would be to initialize this to `nullptr` and get rid of the `avformat_free_context`.)

Reply:
I'd like to merge this as is and then address any unrelated changes in a new PR.